llvm/lib/Transforms/Scalar/LoopIdiomRecognize.cpp

```diff
@@ -126,6 +127,27 @@ static cl::opt<bool, true>
                       cl::location(DisableLIRP::Memcpy), cl::init(false),
                       cl::ReallyHidden);
 
+bool DisableLIRP::Strlen;
+static cl::opt<bool, true>
+    DisableLIRPStrlen("disable-loop-idiom-strlen",
+                      cl::desc("Proceed with loop idiom recognize pass, but do "
+                               "not convert loop(s) to strlen."),
+                      cl::location(DisableLIRP::Strlen), cl::init(false),
+                      cl::ReallyHidden);
+
+/// Some target libraries have a significant call overhead for `wcslen`,
+/// which can degrade performance when the input string is not long enough
+/// to justify the cost. To avoid unnecessary performance penalties,
+/// we disable it by default.
+bool DisableLIRP::Wcslen;
+static cl::opt<bool, true> EnableLIRPWcslen(
+    "enable-loop-idiom-wcslen",
+    cl::desc("Proceed with loop idiom recognize pass, "
+             "enable conversion of loop(s) to wcslen."),
+    cl::location(DisableLIRP::Wcslen), cl::init(true),
+    cl::callback([](const bool &) { DisableLIRP::Wcslen = false; }),
```

Comment:
```suggestion
    "disable-loop-idiom-wcslen",
    cl::desc("Proceed with loop idiom recognize pass, "
             "enable conversion of loop(s) to wcslen."),
    cl::location(DisableLIRP::Wcslen), cl::init(false),
```
The callback is really weird. I assume you want to invert the condition? The storage is called `DisableLIRP::Wcslen`.